scripts/zkai_native_seq32_attention_mlp_median_timing_gate.py

```
@@ -0,0 +1,499 @@
+    expect_equal(payload["target"]["profile_id"], EXPECTED_PROFILE_ID, "target profile", layer="artifact_binding")
+    expect_equal(payload["target"]["typed_bytes"], EXPECTED_TYPED_BYTES, "target typed bytes", layer="artifact_binding")
+    expect_equal(payload["target"]["json_proof_bytes"], EXPECTED_JSON_PROOF_BYTES, "target JSON proof bytes", layer="artifact_binding")
+    expect_equal(payload["timing_summary"]["sample_count"], EXPECTED_SAMPLE_COUNT, "sample count", layer="timing_metrics")
+    if payload["timing_summary"]["prove_median_us"] <= 0 or payload["timing_summary"]["verify_median_us"] <= 0:
+        raise Seq32TimingGateError("timing medians must be positive", layer="timing_metrics")
+    if payload["timing_summary"]["generated_proof_json_bytes_all"] != [EXPECTED_JSON_PROOF_BYTES] * EXPECTED_SAMPLE_COUNT:
+        raise Seq32TimingGateError("generated proof JSON bytes drift", layer="timing_metrics")
+    if payload["host_metadata_policy"]["excluded"] != ["hostname", "username", "absolute_local_paths"]:
+        raise Seq32TimingGateError("host metadata privacy boundary drift", layer="timing_metrics")
+    for source in payload["source_artifacts"]:
+        source = require_object(source, "source artifact", layer="artifact_binding")
```

Ruff (0.15.13)

[warning] `for` loop variable `source` overwritten by assignment target

(PLW2901)